pkg/controller/nodenetworkconfigurationpolicy/conditions/manager.go

```diff
@@ -0,0 +1,79 @@
+package conditions
+
+import (
+	"context"
+	"github.com/go-logr/logr"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+
+	nmstatev1alpha1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1alpha1"
+)
+
+var (
+	log = logf.Log.WithName("policy/conditions/manager")
+)
+
+type Manager struct {
+	client     client.Client
+	nodeName   string
+	policyName types.NamespacedName
+	logger     logr.Logger
+}
+
+func NewManager(client client.Client, nodeName string, policyName types.NamespacedName) Manager {
+	return Manager{
+		client:     client,
+		nodeName:   nodeName,
+		policyName: policyName,
+		logger:     log.WithValues("node", nodeName, "policy", policyName),
+	}
+}
+
+func (m *Manager) Progressing() {
+	err := m.updateEnactmentCondition(setEnactmentProgressing, "Applying desired state")
+	if err != nil {
+		m.logger.Error(err, "Error change state to progressing")
+	}
+}
+func (m *Manager) FailedToConfigure(failedErr error) {
+	err := m.updateEnactmentCondition(setEnactmentFailedToConfigure, failedErr.Error())
+	if err != nil {
+		m.logger.Error(err, "Error chaing state to failing at configure with error: %v", failedErr)
+	}
+}
+
+func (m *Manager) FailedToFindPolicy(failedErr error) {
+	err := m.updateEnactmentCondition(setEnactmentFailedToFindPolicy, failedErr.Error())
+	if err != nil {
+		m.logger.Error(err, "Error changing state to finling at finding policy with error: %v", failedErr)
+	}
+}
+
+func (m *Manager) Success() {
+	err := m.updateEnactmentCondition(setEnactmentSuccess, "successfully reconciled")
+	if err != nil {
+		m.logger.Error(err, "Success condition update failed while reporting success: %v", err)
+	}
+}
+
+func (m *Manager) updateEnactmentCondition(
+	condition func(*nmstatev1alpha1.EnactmentList, string, string),
+	message string,
+) error {
+	// Set enactment condition
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
I think we can get rid of this helper. Due to that we do 3 times as many updates of conditions. We should update all 3 conditions at once and be atomic.

Reply:
the conditionSetter changes all three at the same time so we are already atomic.

Reply:
ah, gotcha

Reply:
Then please make it conditionsSetter so I don't get confused twice